### DISCUSSION TITLE
BTag Tools and "RESHAPE" option

### DISCUSSION THREAD

**Discussion by aebid** [scikit-hep/coffea#767] [General]
Hi experts, I have been trying to use the BTagScaleFactor class to calculate the SF from a CSV file and am running into an issue. From what I understand, there are 4 options, "LOOSE", "MEDIUM", "TIGHT", and "RESHAPE".

According to this line in the eval() function, https://github.com/CoffeaTeam/coffea/blob/65b137fa4ecfe836bb1cae1cbd54e8cef8b9e477/coffea/btag_tools/btagscalefactor.py#L197-L199 the discriminator should be an optional use for all cases but necessary for the "RESHAPE" case. But the discriminator is then ignored for cases not "RESHAPE" https://github.com/CoffeaTeam/coffea/blob/65b137fa4ecfe836bb1cae1cbd54e8cef8b9e477/coffea/btag_tools/btagscalefactor.py#L209-L220

Is this intended or am I confused about something? Looking at my CSV file from the BTagCalibration twiki, the other options should still use the discriminator. Even in the dense_map_lookup dictionary the discriminator edges are ignored. https://github.com/CoffeaTeam/coffea/blob/65b137fa4ecfe836bb1cae1cbd54e8cef8b9e477/coffea/btag_tools/btagscalefactor.py#L167-L180

Thanks!

**Comment by nsmith-** [comment]
In the past the correction algorithm did not depend on the discriminator value for cases other than reshape. It may have changed, but I would be surprised, since with a working point the discriminator effectively becomes a boolean (either greater or less than the WP).
That said, can you use the correctionlib json files for your use case? I was under the impression the csv is being phased out.
